src/java.desktop/unix/native/libawt_xawt/awt/screencast_pipewire.c

```diff
@@ -89,8 +92,8 @@ static void doCleanup() {
         struct ScreenProps *screenProps = &screenSpace.screens[i];
         if (screenProps->data) {
             if (screenProps->data->stream) {
-                fp_pw_thread_loop_lock(pw.loop);
                 fp_pw_stream_disconnect(screenProps->data->stream);
+                fp_pw_thread_loop_lock(pw.loop);
```

Comment:
Moving `pw_stream_disconnect` call out of this lock fixes [JDK-8310334](https://bugs.openjdk.org/browse/JDK-8310334)